### PR TITLE
Infused fire

### PR DESCRIPTION
**Pull request by mitchej123**
Fix Infused Fire (Should no longer destroy servers!)
* Cleanup fire code, fix logic errors, and properly tether to a reasonable distance of the source.  Start with 0 meta (source), which doesn't burn out, and spread (randomly increasing meta) until meta 15 - which no longer spreads.
* Reduce Allocations - Use a static hashmap instead of building it on the fly each time
* Reduce Allocations - Remove BlockTuple() which was (Block, Meta) and was only being used with meta != 0 for Ordo Fire
* Disable Ordo Fire as it doesn't fit NH (ore -> blocks) and wouldn't work with GT ore anyway - Pending new transmutation targets from Alastor or others
* Mixin to fix fire rendering on water -- call the subclass's canCatchFire instead of the base class

### CODE REVIEW

**Comment by mitchej123** [comment]
Main commits to focus on: 
[Cleanup Fire Code](https://github.com/GTNewHorizons/ThaumicTinkerer/pull/31/commits/f7f2645e02dd51abd36aeb6334e50e85abf43068)
[Fewer Allocations
](https://github.com/GTNewHorizons/ThaumicTinkerer/pull/31/commits/13dc33965469b30f775c29f2c4552f3bcd41c09c)[BlockTuple removal & final fire fix](https://github.com/GTNewHorizons/ThaumicTinkerer/pull/31/commits/494be98b52aeec5bcc494da10482e4b88642d523)

File: src/main/java/thaumic/tinkerer/common/block/fire/BlockFireAir.java [hunk 1]
```diff
@@ -51,25 +53,31 @@ public ThaumicTinkererRecipe getRecipeItem() {
...
-        HashMap<BlockTuple, BlockTuple> result = new HashMap<BlockTuple, BlockTuple>();
-        result.put(new BlockTuple(Blocks.log), new BlockTuple(Blocks.sand));
-        result.put(new BlockTuple(Blocks.leaves), new BlockTuple(Blocks.sandstone));
-        result.put(new BlockTuple(Blocks.leaves2), new BlockTuple(Blocks.sandstone));
-        result.put(new BlockTuple(Blocks.log2), new BlockTuple(Blocks.sand));
-        result.put(new BlockTuple(Blocks.ice), new BlockTuple(Blocks.glass));
+    private static void initBlockTransformation() {
+        blockTransformation = new HashMap<>();
+        blockTransformation.put(Blocks.log, Blocks.sand);
+        blockTransformation.put(Blocks.leaves, Blocks.sandstone);
```
Comment (Alastors) [inline src/main/java/thaumic/tinkerer/common/block/fire/BlockFireAir.java:59]: Looks good

File: src/main/java/thaumic/tinkerer/common/block/fire/BlockFireAir.java [hunk 2]
```diff
@@ -51,25 +53,31 @@ public ThaumicTinkererRecipe getRecipeItem() {
...
+        blockTransformation.put(Blocks.leaves2, Blocks.sandstone);
+        blockTransformation.put(Blocks.log2, Blocks.sand);
+        blockTransformation.put(Blocks.ice, Blocks.glass);
         if (ConfigHandler.enableCake) {
```
Comment (Alastors) [inline src/main/java/thaumic/tinkerer/common/block/fire/BlockFireAir.java:63]: Lmfaooo, I appreciate that this is a config for some reason

File: src/main/java/thaumic/tinkerer/common/block/fire/BlockFireEarth.java [hunk 3]
```diff
@@ -55,33 +57,39 @@ public ThaumicTinkererRecipe getRecipeItem() {
...
+        blockTransformation = new HashMap<>();
+        blockTransformation.put(Blocks.sand, Blocks.dirt);
+        blockTransformation.put(Blocks.gravel, Blocks.clay);
+        blockTransformation.put(Blocks.nether_brick, Blocks.planks);
+        blockTransformation.put(Blocks.nether_brick_fence, Blocks.fence);
+        blockTransformation.put(Blocks.nether_brick_stairs, Blocks.oak_stairs);
+        blockTransformation.put(Blocks.cactus, Blocks.log);
+        blockTransformation.put(Blocks.snow_layer, Blocks.tallgrass);
+        blockTransformation.put(Blocks.stone, Blocks.dirt);
+        blockTransformation.put(Blocks.mob_spawner, Blocks.iron_block);
```
Comment (Alastors) [inline src/main/java/thaumic/tinkerer/common/block/fire/BlockFireEarth.java:70]: We might need to put the terra one on the tinker block as well, these are really strange

**Review by Alastors — APPROVED** [review]
Read through all of it, this has made me realize just how useless so many of the infused fires are, in the future I plan to reduce their warp as consequence, likewise, we might strongly need to consider changing the functionality of not only the ordo (as has been discussed), but likewise the terra and possibly other fires

**Pull request merged**